examples/Custom_Hooks/probability_of_improvement.py

```diff
@@ -0,0 +1,219 @@
+## Example for calculating and plotting the Probability of Improvement
+
+# This example demonstrates how to use the
+# {func}`register_hooks <baybe.utils.basic.register_hooks>` utility
+# to calculate and analyse the probability of improvement (PI) evolving during iterations:
+# * We define a hook that is compatible with the general
+#   {meth}`RecommenderProtocol.recommend <baybe.recommenders.base.RecommenderProtocol.recommend>`
+#   interface,
+# * attach it to a recommender,
+# * and watch it take action.
```

Comment:
You copied this pretty much 1:1 from the basic example and this is too generic for here. In the basic example, the specific role of the hooks didn't matter at all, it was all about setting up the workflow. Here, the situation is opposite: the workflow is assumed to be known, but we care about the specific application. So please make a concrete agenda here 🙃

Reply:
What do you think about this:
```suggestion
# This example demonstrates how to use the
# {func}`register_hooks <baybe.utils.basic.register_hooks>` utility
# to calculate and plot the probability of improvement (PI) evolving during iterations:
# * We define a hook that is compatible with the general
#   {meth}`RecommenderProtocol.recommend <baybe.recommenders.base.RecommenderProtocol.recommend>`
#   interface, where we will extract the PI from each iteration
# * define the plotting function,
# * create `my_recommender` and attach the hook to it,
# * generate the parameters used for the `searchspace`,
# * create our `campaign`,
# * perform some recommend iterations,
# * and plot the results in a 3D plot.
```

Reply:
This is too detailed, focusing on elements that are not relevant to make the point. For example, the parameters/searchspace/etc creation is not import here.

How about:

```markdown
This example demonstrates how the
{func}`register_hooks <baybe.utils.basic.register_hooks>` utility
could be used to extract the probability of improvement (PI) from a running campaign:
* We define a hook that is compatible with the
  {meth}`BotorchRecommender.recommend <baybe.recommenders.pure.bayesian.BotorchRecommender.recommend>`
  interface and let's us extract the PI achieved after each experimental iteration,
* attach the hook to the recommender driving our campaign,
* and plot the evolving PI values after campaign completion.
```

Reply:
Sounds good. Will update the description 👍